File: LeetCode/1014/Solution.cpp
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& A) {
        int ret = 0;
        int temp = A[0] + 0;
        
        for(int i=1;i<A.size();++i) {
            ret = max(ret, temp + A[i] - i);
            temp = max(temp, A[i] + i);
        }
        
        return ret;
    }
};

File: LeetCode/1014/Solution.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int left = values[0];
        int result = 0;
        for(int i=1;i<values.length;++i) {
            result = Math.max(result, left+values[i]-i);
            left = Math.max(left, values[i]+i);
        }

        return result;
    }
}
